### ISSUE TITLE
bug: pipe.pkl binary model committed to main branch — binary artifacts must not be version-controlled, causes 40MB+ repo bloat and stale model served to all users without retraining

### ISSUE THREAD

**Issue by divyanshim27** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#605]
## Summary

`pipe.pkl` — the serialized scikit-learn pipeline (OneHotEncoder +
LogisticRegression) — is committed directly to the `main` branch and tracked
by git. Binary ML model artifacts must never be version-controlled. This
creates several serious problems unique to ML projects.

## Why This Is Critical for an ML Project

**1. Stale model served to all users permanently:**
The `pipe.pkl` was serialized with a specific version of scikit-learn.
When any contributor installs a different scikit-learn version (e.g.,
1.3 vs 1.5), `pickle.load(pipe.pkl)` raises:
sklearn.exceptions.InconsistentVersionWarning: Trying to unpickle

estimator LogisticRegression from version 1.3.0 when using version 1.5.2.

Or silently loads a corrupted model producing wrong probabilities — no error,
wrong predictions, no indication to the user.

**2. 40MB+ repository bloat permanently:**
Git tracks binary history. Once a `.pkl` is committed, even if deleted
later, it lives forever in the git object store. With 25 forks, this bloat
is already multiplied across every clone.

**3. No reproducibility:**
There is no `train.py` or `model_training.ipynb` in the repository.
Contributors cannot verify how the model was trained, what data splits
were used, or what the train/test accuracy was. The `dataset_info.txt`
mentions the Kaggle dataset but gives no training script.

**4. `pipe.pkl` in `.gitignore`? No.**
The `.gitignore` has no entry for `*.pkl`.

## Fix Required

```bash
# 1. Remove pipe.pkl from git tracking
git rm --cached pipe.pkl
echo "*.pkl" >> .gitignore
echo "*.joblib" >> .gitignore
git commit -m "chore: remove binary model from version control"
```

**2. Add `train.py` that regenerates `pipe.pkl` from the CSV data:**

```python
# train.py
import pandas as pd
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.linear_model import LogisticRegression
import pickle

matches = pd.read_csv("matches.csv")
deliveries = pd.read_csv("deliveries.csv")
# ... feature engineering per ARCHITECTURE diagram
# ... pipeline fit
with open("pipe.pkl", "wb") as f:
    pickle.dump(pipe, f)
print("Model saved to pipe.pkl")
```

**3. Update README setup section:**

```markdown
## 4 — Train the Model
python train.py
# Generates pipe.pkl locally — not tracked by git
```

## Acceptance Criteria

- [ ] `pipe.pkl` removed from git history using `git rm --cached`
- [ ] `*.pkl` added to `.gitignore`
- [ ] `train.py` added with full reproducible training pipeline
- [ ] README updated with model training step before `streamlit run app.py`
- [ ] `train.py` prints scikit-learn version and final model accuracy

**Labels:** `bug`, `ml`, `chore`, `level: intermediate`

**Comment by divyanshim27** [comment]
Hey @Arnav-Singh-5080   ,
I have deeply gone through the project repo with all-proper local running and testing , and now my observations and analysis , I found this as major lagging/enhancement in the project repo .
Based on my issue understanding and my skill-set for solving this problem, I would want to contribute to the issue.
Please assign it to me .
